VSCode/armsnippets.json

```diff
@@ -506,6 +506,145 @@
+        "                          \"sourceAddresses\": [",
+        "                             \"${16:10.0.0.4}\"",
+        "                          ],",
+        "                          \"destinationAddresses\": [",
+        "                              \"[reference(resourceId('Microsoft.Network/publicIPAddresses', '${34:publicIPAddress1}')).ipAddress]\"",
+        "                          ],",
+        "                          \"destinationPorts\": [",
+        "                              \"${18|3389,1-64000|}\"",
+        "                          ],",
+        "                          \"protocols\": [",
+        "                              \"${19|TCP,UDP,Any,ICMP|}\"",
+        "                          ],",
+        "                          \"translatedAddress\": \"${20:10.0.0.254}\",",
+        "                          \"translatedPort\": \"${21|2500,1-64000|}\"",
+        "                      }",
+        "                   ]",
+        "               }",
+        "           }",
+        "        ],",
+        "       \"networkRuleCollections\": [",
+        "           {",
+        "              \"name\": \"${22:networkCollectionName}\",",
+        "               \"properties\": {",
+        "                   \"priority\": \"${23|100,100-65000|}\",",
+        "                   \"action\": {",
+        "                      \"type\": \"${24|Deny,Allow|}\"",
+        "                   },",
+        "                   \"rules\": [",
+        "                      {",
+        "                          \"name\": \"${25:networkRuleName}\",",
+        "                          \"description\": \"${26:networkRuleDescription}\",",
+        "                          \"sourceAddresses\": [",
+        "                             \"${27:10.0.0.4}\"",
+        "                          ],",
+        "                          \"destinationAddresses\": [",
+        "                              \"${28:10.0.1.4}\"",
+        "                          ],",
+        "                          \"destinationPorts\": [",
+        "                              \"${29|1,1-64000,*|}\"",
+        "                          ],",
+        "                          \"protocols\": [",
+        "                              \"${30|TCP,UDP,Any,ICMP|}\"",
+        "                          ]",
+        "                      }",
+        "                   ]",
+        "               }",
+        "           }",
+        "        ],",
+        "       \"ipConfigurations\": [",
+        "           {",
+        "               \"name\": \"${31:ipConfigName}\",",
+        "               \"properties\": {",
+        "                   \"subnet\": {",
+        "                       \"id\": \"${32:[resourceId('virtualNetworkResourceGroup', 'Microsoft.Network/virtualNetworks/subnets', 'virtualNetworkName', 'AzureFirewallSubnet')]}\"",
```

Comment:
I'd tokenize only the specific params within the call to resourceId() like:
\\"id\\": \\"[resourceId('${99:virtualNetworkResourceGroup}', 'Microsoft.Network/virtualNetworks/subnets', '${99:virtualNetworkName}', '${99:AzureFirewallSubnet}')]\\"",

Reply:
Will do, except I will leave the AzureFirewallSubnet as that is a requirement for AZFW (the subnet has to be named exactly that).

Reply:
Will do, except I will leave the AzureFirewallSubnet as that is a requirement for AZFW (the subnet has to be named exactly that).

Reply:
Eh, and, I think we need different token values there. the VNet name and RG will probably not be the same so we can't use the same number.